components/dnslink-result.vue

```diff
@@ -0,0 +1,188 @@
+            <td></td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+    <div v-if="tab === 'err'">
+      <table>
+        <thead>
+          <tr>
+            <td>type</td>
+            <td>domain</td>
+            <td style="width: 100%">entry</td>
+            <td>reason</td>
+            <td>path</td>
+          </tr>
+        </thead>
+        <tbody>
+          <!-- eslint-disable-next-line vue/require-v-for-key -->
+          <tr v-for="entry in result.log">
+            <td><abbr :title="codeInfo(entry.code)">{{ entry.code }}</abbr></td>
+            <td>{{ entry.domain }}</td>
+            <td>{{ entry.entry }}</td>
+            <td><abbr :title="reasonInfo(entry.reason)">{{ entry.title }}</abbr></td>
+            <td>{{ pathToString(entry) }}</td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+  </div>
+</template>
+<style lang="scss" scoped>
+.error {
+  background-color: var(--c-badge-warning);
+  border-radius: 0.2em;
+  margin-bottom: 0;
+}
+.result {
+  border-radius: 0.2em;
+  overflow: hidden;
+  margin-top: 0.75em;
+  color: #fff;
+  background: var(--code-bg-color);
+  &.inactive {
+    opacity: 0.25;
+  }
+  .tabbar {
+    &, button {
+      background: var(--code-hl-bg-color);
+      border: 0;
+    }
+    button {
+      font-size: 1em;
+      cursor: pointer;
+      color: #fff;
+      padding: .4em .75em;
+      display: inline-block;
+      border-top: 2px solid var(--code-hl-bg-color);
+
+      &.selected {
+        color: var(--c-brand);
+        background: var(--code-bg-color);
+        border-top-color: var(--c-brand);
+      }
+      &:focus {
+        outline: 0;
+        border-top-color: var(--c-brand);
+      }
+    }
+  }
+  table {
+    min-width: 100%;
+    margin: 0;
+    tr {
+      background-color: var(--c-hl-bg-color) !important;
+      border: 1px solid var(--code-hl-bg-color);
+    }
+    td {
+      white-space: nowrap;
+      border: 0;
+    }
+    thead td {
+      color: rgba(255, 255, 255, 0.55);
+    }
+  }
+}
+</style>
+<script lang="ts">
+import { PathEntry } from '@dnslink/js'
+import { defineComponent, ref, toRefs } from 'vue'
+export default defineComponent({
+  props: {
+    result: {
+      type: Object,
+      result: null
+    }
+  },
+  setup: props => ({
+    ...toRefs(props),
+    tab: ref<'out' | 'err'>('out')
+  }),
+  methods: {
+    pathToString (path: PathEntry): string {
+      let result = path.pathname !== undefined ? path.pathname : ''
+      let sep = '?'
+      if (path.search) {
+        for (const [key, values] of Object.entries(path.search)) {
+          for (const value of values) {
+            result += sep + encodeURIComponent(key) + '=' + encodeURIComponent(value)
+            sep = '&'
+          }
+        }
+      }
+      return result
+    },
+    reasonInfo (reason: string): string {
+      if (reason === 'NO_VALUE') {
+        return 'An dnslink entry needs to have a value, like: dnslink=/key/value.'
+      }
```

Comment:
nit: probably want to avoid so many IFs and use  dict instead:
```js
const reasons = {
  "NO_VALUE": "An dnslink entry needs to have a value, like: dnslink=/protocol/path.",
  "WRONG_START": "A dnslink entry needs to start with a /.",
  ...
}
// and then acces them via 
return  reasons[reason]
```

Reply:
Thanks! I am thinking about moving this to `dnslink/js`.

Reply:
Added to https://github.com/dnslink-std/js/commit/2230f04a4885c3df37339caa0b16725df42867ef and used in 55e022daae9de3b59aee06966d509dc713b5439c